packages/one-app-bundler/webpack/loaders/common.js

```diff
@@ -39,30 +39,72 @@ const cssLoader = ({ name = '', importLoaders = 2 } = {}) => ({
 
 /* eslint-disable inclusive-language/use-inclusive-words --
 config options for a third party library */
-const purgeCssLoader = () => {
-  const configOptions = getConfigOptions();
-  const whitelistPatterns = configOptions.purgecss.whitelistPatterns
-    ? configOptions.purgecss.whitelistPatterns.map((pattern) => new RegExp(pattern, 'i'))
+// transform strings deep and greedy parameters into regex
+const reconcileSafeList = (safelist) => {
+  if (!safelist || Array.isArray(safelist)) {
+    return safelist;
+  }
+  const configOptionsReconciled = safelist;
+
+  const greedy = safelist.greedy
+    ? safelist.greedy.map((pattern) => new RegExp(pattern, 'i'))
     : [];
-  const whitelistPatternsChildren = configOptions.purgecss.whitelistPatternsChildren
-    ? configOptions.purgecss.whitelistPatternsChildren.map((pattern) => new RegExp(pattern, 'i'))
+  configOptionsReconciled.greedy = greedy;
+  const deep = safelist.deep
+    ? safelist.deep.map((pattern) => new RegExp(pattern, 'i'))
     : [/:global$/];
-  if (configOptions.purgecss.disabled) return [];
+  configOptionsReconciled.deep = deep;
+
+  return configOptionsReconciled;
+};
+
+const purgeCssLoader = () => {
+  const { purgecss } = getConfigOptions();
+  const safelist = reconcileSafeList(purgecss.safelist);
+  if (purgecss.disabled) return [];
+  let aggregatedStandard = [];
+  let safelistDeep = [/:global$/];
+  // aggregate the various whitelist options if safelist is not present
+  if (!safelist) {
```

Comment:
can this be known based on `purgecss.safelist`? or the could the `reconcileSafeList` output a falsy value?

Reply:
I see, but it could be moved to improve readability, its value is being check already at line 63, we could do something like:

```js
let { safelist } = purgecss

if (safelist && !Array.isArray(safelist)) {
   safelist = reconcileSafeList(purgecss.safelist);
} else {
   let aggregatedStandard = [];
   let safelistDeep = [/:global$/];
   if (purgecss.whitelistPatterns) {
....
}

```

Reply:
```let aggregatedStandard = [];   let safelistDeep = [/:global$/];``` these two still need to be initialized outside of the original if statement due to scope

Reply:
true, my bad, they need to be declared before the if statement